### PR TITLE
Entries: Promote jQuery.validator.methods property as public api

### PR DESCRIPTION
**Pull request by staabm**
Includes a example which shows how to make built-in email validation use
a custom email regular expression pattern.

### CODE REVIEW

File: entries/jQuery.validator.methods.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<?xml-stylesheet type="text/xsl" href="../entries2html.xsl" ?>
+<entry type="method" name="jQuery.validator.methods" returns="Object">
+	<title>jQuery.validator.methods</title>
+	<desc>Object holding all validation methods known to the validator</desc>
+	<example>
+		<desc>Sets a custom email pattern for the built-in email validation rule.</desc>
+		<code><![CDATA[
+$.validator.methods.email = function( value, element ) { 
+  return this.optional( element ) || /whatEverRegexILike/.test( value );
```
Comment (jzaefferer) [inline entries/jQuery.validator.methods.xml]: This should be an actual email regex, though a very simple one.

File: entries/jQuery.validator.methods.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<?xml-stylesheet type="text/xsl" href="../entries2html.xsl" ?>
+<entry type="method" name="jQuery.validator.methods" returns="Object">
+	<title>jQuery.validator.methods</title>
+	<desc>Object holding all validation methods known to the validator</desc>
```
Comment (jzaefferer) [inline entries/jQuery.validator.methods.xml]: Amend: "This can be accessed to override individual methods, while keeping the default messages."

**Comment by jzaefferer** [comment]
There needs to be a link to this page somewhere on the site. At least on `pages/documentation.md`.

**Comment by staabm** [comment]
adjusted per feedback.

File: entries/email-method.xml [hunk 3]
```diff
@@ -5,6 +5,7 @@
 	<longdesc>
 		Return true if the value is a valid email address.
 		<p>Works with text inputs.</p>
+        <p>In case you need a custom email pattern use the <a href="/jQuery.validator.methods/">$.validator.methods property</a>.</p>
```
Comment (jzaefferer) [inline entries/email-method.xml]: Please fix the indent, mixing spaces and tabs here. Also in the next diff.

File: entries/jQuery.validator.methods.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<?xml-stylesheet type="text/xsl" href="../entries2html.xsl" ?>
+<entry type="method" name="jQuery.validator.methods" returns="Object">
+	<title>jQuery.validator.methods</title>
+	<desc>
+        Object holding all validation methods known to the validator.
+        This can be accessed to override individual methods, while keeping the default messages.</desc>
```
Comment (jzaefferer) [inline entries/jQuery.validator.methods.xml]: The linebreak will collapse when rendered in the browser. If that's okay, it shouldn't be in the source. If its not okay, there should be explicit markup to render separate lines. (also mixed indent)

File: entries/jQuery.validator.methods.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	<example>
+		<desc>Sets a custom email pattern for the built-in email validation rule.</desc>
+		<code><![CDATA[
+$.validator.methods.email = function( value, element ) {
```
Comment (jzaefferer) [inline entries/jQuery.validator.methods.xml]: `git diff` shows trailing whitespace here :/

**Comment by jzaefferer** [comment]
Thanks for the updates. Content looks good, issues should be easy enough to address. I recommend adjusting your editor to show indent/whitespace by default and removing trailing whitespace on save. Dunno what you use, in Sublime Text its `"draw_white_space": "all",` and `"trim_trailing_white_space_on_save": true,`.

**Comment by staabm** [comment]
sorry for the ugly whitespaces... usually I am using Zend Studio which resolves such issues, but I made this changes using Notepad++ .. will try if it is configurable in Notepad++

**Pull request merged**